lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -1188,6 +1187,48 @@ push @tests, __PACKAGE__->new(
   max_version => '3.0'
 );
 
+push @tests, __PACKAGE__->new(
+    test_query => q(SELECT ac.lsmb_entry_id, ac.trans_id, ac.id, ac.chart_id, ac.memo, ac.amount, xx.description,
+                          ch.description as account, ch.accno, ch.link, ch.charttype, ch.category, ac.cleared, approved
+                     FROM acc_trans ac
+                     JOIN (
+                               SELECT g.id, g.description FROM gl g
+                         UNION SELECT a.id, n.name        FROM ar a JOIN customer n ON n.id = a.customer_id
+                         UNION SELECT a.id, n.name        FROM ap a JOIN vendor n   ON n.id = a.vendor_id
+                     ) xx ON xx.id = ac.trans_id
+                     JOIN chart ch ON (ac.chart_id = ch.id)
+                    WHERE ( ch.category NOT IN ( 'A', 'L', 'Q' )
+                         OR ch.link NOT LIKE '%paid' )
+                      AND ac.cleared IS NOT NULL
+                      AND ac.approved
+                 ORDER BY trans_id, ac.id, accno, transdate),
+  display_name => marktext('Unneeded Reconciliations'),
+          name => 'reconciliation_on_unrelated_accounts',
+  display_cols => ['trans_id', 'id', 'memo', 'amount', 'description',
+                   'accno', 'account', 'link', 'category', 'cleared', 'approved'],
+       columns => ['cleared'],
+    id_columns => ['lsmb_entry_id'],
+  instructions => marktext(
+                   'Reconciliations should be on asset, liability or equity accounts only.<br>
+Please review the dates and categories in the original application if needed'),
```

Comment:
why "if needed"? Is this a request to do so (and thus a validation failure) or not?

Reply:
They may be able to do all in LSMB or go back to SL to fix it.

Reply:
maybe instead of "if needed" -> "or click 'Skip' to continue" ?

Reply:
I don't want to promote Skip. It is fixable and should be.